src/renderer/contexts/InstantOnboardingContext.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react'
+
+import type { PropsWithChildren } from 'react'
+import type { QrWithUrl } from '../backend/qr'
+
+type InstantOnboardingContextValue = {
+  setShowInstantOnboarding: (value: boolean) => void
+  setWelcomeQr: (value?: QrWithUrl) => void
+  showInstantOnboarding: boolean
+  welcomeQr?: QrWithUrl
+}
+
+export const InstantOnboardingContext =
+  React.createContext<InstantOnboardingContextValue | null>(null)
+
+export const InstantOnboardingProvider = ({
+  children,
+}: PropsWithChildren<{}>) => {
+  // Flag to display "Instant Onboarding" UI in welcome screen
+  const [showInstantOnboarding, setShowInstantOnboarding] = useState(false)
```

Comment:
I can read/understand two different things from the comment:
- instant onboarding option is shown or hidden in welcome screen
- welcome screen gets replaced by onboarding screen

solution idea:
- change var name and/or change comment to clarify what it does

Reply:
> I can read/understand two different things from the comment:
> 
>     * instant onboarding option is shown or hidden in welcome screen
>     * welcome screen gets replaced by onboarding screen

This sounds to me like an implementation detail this state variable doesn't need to know about ..

Reply:
I find it hard to come up with something else here without any clear suggestion _how_ you understand this part of code - ("misunderstandings" are well .. misunderstandings, so any input would be helpful to improve). Feel free, to do the changes directly.

My nitpick take on your comment: This component doesn't need to be super-aware of _how_ it is used in the UI, it just needs to handle this flag which makes something visible or not.

Reply:
I changed the comment in 6919869875642558c42106bbdd50a1a77aef0fe9